### PR TITLE
[ODS-3931] Add versioning check for plugin extensions

### PR DESCRIPTION
**Pull request by tnunnerydl**

### CODE REVIEW

File: Application/EdFi.Ods.Common/Models/Validation/ApiModelExistsValidator.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using FluentValidation;
+
+namespace EdFi.Ods.Common.Models.Validation
+{
+    public class ApiModelExistsValidator : AbstractValidator<string>
+    {
+
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Models/Validation/ApiModelExistsValidator.cs:15]:
```suggestion

```

File: Application/EdFi.Ods.Common/Models/Validation/ApiModelExistsValidator.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                .Must(
+                    e =>
+                    {
+                        var metaDataFolder = Path.Combine(e, "Artifacts", "Metadata");
+                        var files = Directory
+                            .GetFiles(metaDataFolder)
+                            .Where(s => regex.IsMatch(s))
+                            .ToList();
+
+                        return files.Count > 0;
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Models/Validation/ApiModelExistsValidator.cs:30]:
```suggestion
                        return Directory
                            .GetFiles(metaDataFolder)
                            .Any(s => regex.IsMatch(s));
```

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        {
+            var regex = new Regex(@"ApiModel.*.json");
+
+            _apiVersion = apiVersion;
+
+            RuleFor(extensionFolder => extensionFolder)
+                .Must(
+                    e =>
+                    {
+                        var metaDataFolder = Path.Combine(e, "Artifacts", "Metadata");
```
Comment (jamessmckay) [inline Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs:29]: we probably need some logging here. I think the intent is that we have an extension and that the model is valid. should be testing both of these rules as  compound?

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs (hunk 3 above)
Comment (tnunnerydl) [reply Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs:29]: We are testing both conditions and logging is done in AssemblyLoaderHelper.

Comment (tnunnerydl) [reply Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs:29]: Actually, I'm not checking both conditions, only validating version here.  There's another validator that checks for extensions.

Comment (tnunnerydl) [reply Application/EdFi.Ods.Common/Models/Validation/IsApiVersionValidValidator.cs:29]: I'm not sure whey we need to make a compound validator here.  Both IsApiVersionValidValidator and IsExtensionPluginValidator are called from AssemblyLoaderHelper.  Combining these two validators into two rules in one validator will only check for valid extension assemblies.  Making the validators more granular allows us to check if the version is valid for extension, profiles and composites, right?

**Review by jamessmckay — APPROVED** [review]

**Pull request merged**